src/rngs/os.rs

```diff
@@ -1102,6 +1102,76 @@ mod imp {
     }
 }
 
+#[cfg(all(target_arch = "wasm32",
+          not(target_os = "emscripten"),
+          not(feature = "stdweb"),
+          feature = "wasm-bindgen"))]
+mod imp {
+    use core::fmt;
+
+    use __wbg_shims::*;
+
+    use {Error, ErrorKind};
+    use super::OsRngImpl;
+
+    #[derive(Clone)]
+    pub enum OsRng {
+        Node(NodeCrypto),
+        Browser(BrowserCrypto),
+    }
+
+    impl OsRngImpl for OsRng {
+        fn new() -> Result<OsRng, Error> {
+            if this.window().is_undefined() {
+                return Ok(OsRng::Node(node_require("crypto")))
+            }
+            let crypto = this.crypto();
+            if crypto.is_undefined() {
+                let msg = "window.crypto is undefined";
+                return Err(Error::new(ErrorKind::Unavailable, msg))
+            }
+            let crypto: BrowserCrypto = crypto.into();
+            if crypto.get_random_values_fn().is_undefined() {
+                let msg = "crypto.getRandomValues is undefined";
```

Comment:
Should be "window.crypto.getRandomValues is undefined"?

Either way these aren't very user-friendly messages, though I guess they may not need to be.